### PR TITLE
refactor: middleware

### PR DESCRIPTION
**Pull request by Karitham**
This PR refactors the middleware package in such a way that promotes non-breaking changes later on, thus, we could now easily support #79 (presented we actually want that and get to it)

It is also slightly safer, since users can't change the options at runtime, which could have been a tiny issue if it ever happened

Change for REST API consumer: close to none (now doesn't return 500 in
case of multiple errors)
Change for dev: Slight function signature change

BREAKING: Middleware function signature have changed (as well as their
names)
FIX: fixes unknown errors due to multierror

### CODE REVIEW

File: middleware/oapi_validate.go [hunk 1]
```diff
@@ -75,67 +90,60 @@ func OapiRequestValidatorWithOptions(swagger *openapi3.T, options *Options) func
...
 			// Split up the verbose error by lines and return the first one
 			// openapi errors seem to be multi-line with a decent message on the first
-			errorLines := strings.Split(e.Error(), "\n")
+			errorLines := strings.Split(err.Error(), "\n")
 			return http.StatusBadRequest, fmt.Errorf(errorLines[0])
-		case *openapi3filter.SecurityRequirementsError:
+		case errors.As(err, &secError):
 			return http.StatusUnauthorized, err
-		default:
+		case errors.As(err, &otherErrr):
```
Comment (jolheiser) [inline middleware/oapi_validate.go:139]: Does this work as-is?
`reqError := &openapi3filter.RequestError{}`
`errors.As(..., &reqError)`

Comment (Karitham) [reply middleware/oapi_validate.go:139]: Yes, for some reason we need double pointer to errors, so I need to instantiate them somewhere, I can't think of anything else that would avoid it.

Comment (jolheiser) [reply middleware/oapi_validate.go:139]: Oh I see, openapi is returning pointer errors.

File: middleware/oapi_validate.go [hunk 2]
```diff
@@ -1,12 +1,9 @@
-// Package middleware implements middleware function for go-chi or net/http,
-// which validates incoming HTTP requests to make sure that they conform to the given OAPI 3.0 specification.
-// When OAPI validation failes on the request, we return an HTTP/400.
```
Comment (jolheiser) [inline middleware/oapi_validate.go:3]: Why was this package comment removed?

Comment (Karitham) [reply middleware/oapi_validate.go:3]: I actually rewrote the package, then reduced the diff, forgot about the comment

File: middleware/oapi_validate.go [hunk 3]
```diff
@@ -1,12 +1,12 @@
 // Package middleware implements middleware function for go-chi or net/http,
 // which validates incoming HTTP requests to make sure that they conform to the given OAPI 3.0 specification.
-// When OAPI validation failes on the request, we return an HTTP/400.
+// When OAPI validation failes on the request, we return an HTTP/400.[refactor/middleware 7ad632e] diff: revert package doc comment deletion
```
Comment (jolheiser) [inline middleware/oapi_validate.go:3]:
```suggestion
// When OAPI validation failes on the request, we return an HTTP/400.
```
sus

**Review by jolheiser — APPROVED** [review]

**Pull request merged**